### DISCUSSION TITLE
Limit the initial available plugins in the plugin drawer?

### DISCUSSION THREAD

**Discussion by ibaasch-evorhei** [react-page/react-page#990] [Q&A] [answered]
Is there a way to limit the initial available plugins in the plugin drawer 
(e.g. when clicking on the "Add blocks" button)? 

Like you can limit the available plugins with 
the property "cellPlugins" in a CellPlugin?

**Comment by macrozone** [comment, marked as answer]
@ibaasch-evorhei 

just pass the cellPlugins you want to show in `cellPlugins` (on the editor).  Is there a reason you want to pass a cellPlugin there that you don't want to show? If you declare "sub plugins" via CellPlugin's cellPlugins property, you don't have to repeat them on the `<Editor />` itself

you can also set `hideInMenu: true` on a cell plugin (see https://github.com/react-page/react-page/blob/master/packages/editor/src/core/types/plugins.ts#L197) to hide it in the menu, if for whatever reason you want to hide them

**Reply by ibaasch-evorhei** [reply]
we have the following scenario:

- empty editor
  - plugin x
  - plugin a
    - plugin x
    - plugin b
      - plugin x
      - plugin c
        - plugin x

                   
if i got you right, then i would have to set it like this?

editor.cellPlugins = [x, a]
a.cellPlugins = [x, b]
b.cellPlugins = [x, c]
c.cellPlugins = [x]
x.cellPlugins = []

**Reply by macrozone** [reply]
i think so yes, 

i am surprised by the depth, can you tell the use case (just out of curiosity)

**Reply by ibaasch-evorhei** [reply]
I have tried your suggestion. At first glance it seems to work. 

In the drawer of plugin b the plugins "x and c" should appear. However, "x and b" appear?

**Reply by ibaasch-evorhei** [reply]
if I specify 
editor.cellPlugins = [x, a, b, c] 
everything works correctly (except for the initial constraint)

**Reply by macrozone** [reply]
ok, looks like a bug, i'll take a look

**Reply by ibaasch-evorhei** [reply]
thank you marco, for looking. 
and thank you for this great editor, which is very helpful for us!

**Reply by macrozone** [reply]
@ibaasch-evorhei i pushed the fix, sorry for the inconvenience!

it should be available under 2.2.1 in 10-20min

<img width="1812" alt="Bildschirmfoto 2021-05-11 um 18 51 01" src="https://user-images.githubusercontent.com/1972353/117854515-d7758900-b289-11eb-95d2-34f058483fbd.png">

thank you for the kind words!

**Reply by macrozone** [reply]
@ibaasch-evorhei its published, i also added your example: https://react-page.github.io/examples/nestedPlugins

**Reply by ibaasch-evorhei** [reply]
thanks for the fix @macrozone! 
i will test it tomorrow.

**Reply by ibaasch-evorhei** [reply]
very nice, it works!

maybe you can mention in the docs that it is not necessary 
to pass all used plugins in `Editor.cellPlugins`.

**Comment by ibaasch-evorhei** [comment]
in our application it should be possible for the user to create a document. 
in this document it should be possible to divide the content hierarchically into pages, sections, subsections, etc.
